src/main/java/com/treasure/hunt/view/utils/TreeConstructor.java

```diff
@@ -0,0 +1,155 @@
+package com.treasure.hunt.view.utils;
+
+import com.treasure.hunt.game.Turn;
+import com.treasure.hunt.jts.geom.GeometryAngle;
+import com.treasure.hunt.strategy.geom.GeometryItem;
+import com.treasure.hunt.strategy.geom.GeometryStyle;
+import com.treasure.hunt.strategy.geom.GeometryType;
+import com.treasure.hunt.strategy.hint.Hint;
+import com.treasure.hunt.strategy.hint.impl.AngleHint;
+import com.treasure.hunt.strategy.hint.impl.CircleHint;
+import com.treasure.hunt.strategy.searcher.SearchPath;
+import javafx.scene.control.TreeItem;
+import javafx.scene.control.TreeView;
+import org.locationtech.jts.geom.Coordinate;
+
+public class TreeConstructor {
+
+    public static TreeView<String> getTree(Turn turn) {
+        TreeView<String> stringTreeView = new TreeView<>(
+                TreeConstructor.createItem(turn)
+        );
+        stringTreeView.setShowRoot(false);
+        return stringTreeView;
+    }
+
+    public static TreeItem<String> createItem(Turn turn) {
+        final TreeItem<String> root = createItem("Turn");
+        root.setExpanded(true);
+
+        if (turn.getTreasureLocation() != null) {
+            root.getChildren().add(createItem("Treasure: %s", print(turn.getTreasureLocation().getCoordinate())));
+        }
+
+        if (turn.getHint() != null) {
+            root.getChildren().add(createItem(turn.getHint()));
+        }
+
+        if (turn.getSearchPath() != null) {
+            root.getChildren().add(createItem(turn.getSearchPath()));
+        }
+
+        return root;
+    }
+
+    public static TreeItem<String> createItem(Hint hint) {
+        final TreeItem<String> root = createItem("Hint");
+        if (hint instanceof AngleHint) {
+            root.setValue("AngleHint");
+            AngleHint angleHint = (AngleHint) hint;
+            TreeItem<String> angleItem = createItem(angleHint.getGeometryAngle());
+            root.getChildren().add(angleItem);
+        }
+        if (hint instanceof CircleHint) {
+            root.setValue("CircleHint");
+            CircleHint circleHint = (CircleHint) hint;
+
+            root.getChildren().addAll(
+                    createItem("center: %s", print(circleHint.getCenter().getCoordinate())),
+                    createItem("radius: %s", circleHint.getRadius())
+            );
+        }
+
```

Comment:
What about other hints maybe use the toString() method

Reply:
a hint could be printed out as a pure string, I will use toString() as fallback, but if there are multiple objects in the hint (like in circle hint) then I wanted to separate these properties thus creating two separate tree items